src/v/kafka/server/partition_proxy.cc

```diff
@@ -29,5 +29,13 @@ std::optional<partition_proxy> make_partition_proxy(
     }
     return std::nullopt;
 }
+std::optional<partition_proxy> make_partition_proxy(
```

Comment:
> kafka/partition_proxy: support ntp creation

why do we need this?

Reply:
Sorry this should be `kafka/partition_proxy: add replicate methods` I'll fix

Reply:
it's probably the same question, tho. why is `ktp` not enough?

Reply:
I think we can stick with `ktp` as the produce path assumes this.

I think I was putting the cart in front of the horse and thinking about supporting internal namespaces (like `redpanda`) to use for wasm binaries so nothing is exposed in a public interface, but I will remove that for now as the rpc doesn't support that at the moment anyway.